# Inferential Statistics:

Inferential statistics is used to gather some information about the population from the sample of data collected.<br>
There are many tests which are used in Inferential statistics depending on the objective and the data.<br>

1. To find out ; Is there a significant difference in the marks of students between section A and section B. For any objective, we formulate a hypothesis.<br><br>

Null Hypothesis (Ho):<br>
There is no significant difference in the marks between students of section A and section B.<br><br>

Alternative Hypothesis (Ha):<br>
There is a significant difference in the marks of section A and section B.<br><br>


# The Test to be used:

Step 1: Check the normality separately for the variables Wing A and Wing B.<br>
Step 2: If both the variables have **normal distribution**, then we use a parametric test known as **independent sample t-test**.<br>
Step 3: If the variables don't have a **normal distribution** then we use a non-parametric test known as **Mann Whitney U test**.<br>

## How to do Independent sample t-test in Python?
import the library and inport the dataset into Python.<br>

To do the independent sample t-test in Python, the function used is **ttest_ind**.<br>

### How to import a specific function from a library
To import the function **ttest_ind** from the Python library **scipy.stats**.

In [1]:
from scipy.stats import ttest_ind

### How to implement "independent sample t-test"?

There are two variables **Wing A** and **Wing B**<br>
The independent sample test will give a "t" value and this **t-value** has to be compared with the **t-critical** value<br>
The code to implement the independent sample t-test:

In [4]:
import pandas as pd
df = pd.read_csv('./students.csv')
df

WingA  WingB
0   10.70   7.20
1    9.89   6.68
2   11.83   9.29
3    9.04   8.95
4    9.37   6.61
5   11.68   8.53
6    8.36   8.92
7    9.76   7.95
8   13.67   7.57
9    8.96   6.38
10   9.51   8.89
11  10.85  10.03
12  10.57   9.30
13  11.06   5.28
14   8.91   9.23
15  11.79   9.25
16  10.59   8.44
17   9.13   6.57
18  12.37  10.61
19   9.91   6.77

In [20]:
t_statistics, p_value = ttest_ind( df['WingA'], df['WingB'])
t_statistics

5.16151166403543

The value from Python we get is the **t-calculated**, this value has to be compared with the **critical value**. This **critical value** will be obtained from t-table.

Use the column for 5 % significance level.<br>
The **dof**= n-2, where **n** is the sample size.<br>
Get the critical value from the statistical table.<br>

### The Interpretation
If the calculated from Python is less than the critical value from the table, then you accept the Ho hypothesis  and conclude there is no difference between the two sections.<br><br>

If the calculated value from Python is greater than the critical from the table, then the Ha is accepted and we conclude there is a difference in the performance of the students between the two sections.<br>

# Checking the Normality for "Wing A"

In [13]:
df.describe()

WingA      WingB
count  20.000000  20.000000
mean   10.397500   8.122500
std     1.370032   1.417195
min     8.360000   5.280000
25%     9.310000   6.747500
50%    10.240000   8.485000
75%    11.215000   9.235000
max    13.670000  10.610000

## Criteria 1:

In [15]:
freq_table = pd.crosstab(df['WingA'],'count')
freq_table

col_0  count
WingA       
8.36       1
8.91       1
8.96       1
9.04       1
9.13       1
9.37       1
9.51       1
9.76       1
9.89       1
9.91       1
10.57      1
10.59      1
10.70      1
10.85      1
11.06      1
11.68      1
11.79      1
11.83      1
12.37      1
13.67      1

In [ ]:
x1 = 10.397500 - 1.370032
x2 = 10.397500 + 1.370032

print("x1=",x1)
print("x2=",x2)

x1= 9.027468
x2= 11.767532000000001


**The counted values are 13, therefore the area is <br>
Area = 13/20 = 0.65**

## Criteria 2:

In [16]:
freq_table = pd.crosstab(df['WingA'],'count')
freq_table

col_0  count
WingA       
8.36       1
8.91       1
8.96       1
9.04       1
9.13       1
9.37       1
9.51       1
9.76       1
9.89       1
9.91       1
10.57      1
10.59      1
10.70      1
10.85      1
11.06      1
11.68      1
11.79      1
11.83      1
12.37      1
13.67      1

In [7]:
x1 = 10.397500 - 2*1.370032
x2 = 10.397500 + 2*1.370032

print("x1=",x1)
print("x2=",x2)

x1= 7.657436000000001
x2= 13.137564000000001


**The counted values are 19, therefore the area is <br>\
Area = 19/20 = 0.95**


## Criteria 2:

In [17]:
freq_table = pd.crosstab(df['WingA'],'count')
freq_table

col_0  count
WingA       
8.36       1
8.91       1
8.96       1
9.04       1
9.13       1
9.37       1
9.51       1
9.76       1
9.89       1
9.91       1
10.57      1
10.59      1
10.70      1
10.85      1
11.06      1
11.68      1
11.79      1
11.83      1
12.37      1
13.67      1

In [9]:
x1 = 10.397500 - 3*1.370032
x2 = 10.397500 + 3*1.370032

print("x1=",x1)
print("x2=",x2)

x1= 6.287404000000001
x2= 14.507596


**The counted values are 20, therefore the area is <br>
Area = 20/20 = 1**

## Conclusion:
The Wing A variable is nearly normally distributed.

# Checking the Normality for Wing B

In [10]:
df.describe()

WingA      WingB
count  20.000000  20.000000
mean   10.397500   8.122500
std     1.370032   1.417195
min     8.360000   5.280000
25%     9.310000   6.747500
50%    10.240000   8.485000
75%    11.215000   9.235000
max    13.670000  10.610000

## Criteria 1:

In [18]:
freq_table = pd.crosstab(df['WingB'],'count')
freq_table

col_0  count
WingB       
5.28       1
6.38       1
6.57       1
6.61       1
6.68       1
6.77       1
7.20       1
7.57       1
7.95       1
8.44       1
8.53       1
8.89       1
8.92       1
8.95       1
9.23       1
9.25       1
9.29       1
9.30       1
10.03      1
10.61      1

In [12]:
x1 = 8.122500 - 1.417195
x2 = 8.122500 + 1.417195

print("x1=",x1)
print("x2=",x2)

x1= 6.705305000000001
x2= 9.539695


**The counted values are 13, therefore the area is <br>
Area = 13/20 = 0.65**

## Criteria 2:

## Looking for "t-value" from Python :

In [19]:
t_statistics, p_value = ttest_ind( df['WingA'], df['WingB'])
t_statistics

5.16151166403543

The t-value from Python is 5.162 , and the critical value from the table is 2.101, 
therefore the alternate hypothesis is accepted which means threre is a significant difference in the performance of students 
between the two sections.

In [ ]:
## How to create a dataframe with Python
## Mann Whitney U test
## paired sample t-test